File: test/Models/energybalance_pkg/src/simplace/Netradiation.java
import  java.io.*;
import  java.util.*;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import javafx.util.*;
import java.time.LocalDateTime;
import java.time.LocalDateTime;
import net.simplace.sim.model.FWSimComponent;
import net.simplace.sim.util.FWSimVarMap;
import net.simplace.sim.util.FWSimVariable;
import net.simplace.sim.util.FWSimVariable.CONTENT_TYPE;
import net.simplace.sim.util.FWSimVariable.DATA_TYPE;
import org.jdom.Element;


public class Netradiation extends FWSimComponent
{
    private FWSimVariable<Double> albedoCoefficient;
    private FWSimVariable<Double> stefanBoltzman;
    private FWSimVariable<Double> elevation;
    private FWSimVariable<Double> minTair;
    private FWSimVariable<Double> maxTair;
    private FWSimVariable<Double> solarRadiation;
    private FWSimVariable<Double> vaporPressure;
    private FWSimVariable<Double> extraSolarRadiation;
    private FWSimVariable<Double> netRadiation;
    private FWSimVariable<Double> netOutGoingLongWaveRadiation;

    private Netradiation(String aName, HashMap<String, FWSimVariable<?>> aFieldMap, HashMap<String, String> aInputMap, Element aSimComponentElement, FWSimVarMap aVarMap, int aOrderNumber)
    {
        super(aName, aFieldMap, aInputMap, aSimComponentElement, aVarMap, aOrderNumber);
    }

    public Netradiation(){
        super();
    }
    @Override
    protected void process()
    {
        double tminTair = minTair.getValue();
        double tmaxTair = maxTair.getValue();
        double talbedoCoefficient = albedoCoefficient.getValue();
        double tstefanBoltzman = stefanBoltzman.getValue();
        double televation = elevation.getValue();
        double tsolarRadiation = solarRadiation.getValue();
        double tvaporPressure = vaporPressure.getValue();
        double textraSolarRadiation = extraSolarRadiation.getValue();
        double tnetRadiation = netRadiation.getValue();
        double tnetOutGoingLongWaveRadiation = netOutGoingLongWaveRadiation.getValue();
        double Nsr;
        double clearSkySolarRadiation;
        double averageT;
        double surfaceEmissivity;
        double cloudCoverFactor;
        double Nolr;
        Nsr = (1.0d - talbedoCoefficient) * tsolarRadiation;
        clearSkySolarRadiation = (0.75d + (2 * Math.pow(10.0d, -5) * televation)) * textraSolarRadiation;
        averageT = (Math.pow(tmaxTair + 273.16d, 4) + Math.pow(tminTair + 273.16d, 4)) / 2.0d;
        surfaceEmissivity = 0.34d - (0.14d * Math.sqrt(tvaporPressure / 10.0d));
        cloudCoverFactor = 1.35d * (tsolarRadiation / clearSkySolarRadiation) - 0.35d;
        Nolr = tstefanBoltzman * averageT * surfaceEmissivity * cloudCoverFactor;
        tnetRadiation = Nsr - Nolr;
        tnetOutGoingLongWaveRadiation = Nolr;
        netRadiation.setValue(tnetRadiation, this);
        netOutGoingLongWaveRadiation.setValue(tnetOutGoingLongWaveRadiation, this);
    }

    @Override
    protected void init()
    {
    }

    @Override
    protected FWSimComponent clone(FWSimVarMap aVarMap)
    {
        return new Netradiation(iName, iFieldMap, iInputMap, iSimComponentElement, aVarMap, iOrderNumber);
    }

    @Override
    public HashMap<String, FWSimVariable<?>> createVariables()
    {
        addVariable(FWSimVariable.createSimVariable("calbedoCoefficient", "albedo Coefficient", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1, 0.23, this));
        addVariable(FWSimVariable.createSimVariable("cstefanBoltzman", "stefan Boltzman constant", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1, 4.903E-09, this));
        addVariable(FWSimVariable.createSimVariable("celevation", "elevation", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", -500, 10000, 0, this));
        addVariable(FWSimVariable.createSimVariable("iminTair", "minimum air temperature", DATA_TYPE.DOUBLE, CONTENT_TYPE.input,"http://www.wurvoc.org/vocabularies/om-1.8/", -30, 45, 0.7, this));
        addVariable(FWSimVariable.createSimVariable("imaxTair", "maximum air Temperature", DATA_TYPE.DOUBLE, CONTENT_TYPE.input,"http://www.wurvoc.org/vocabularies/om-1.8/", -30, 45, 7.2, this));
        addVariable(FWSimVariable.createSimVariable("isolarRadiation", "solar Radiation", DATA_TYPE.DOUBLE, CONTENT_TYPE.input,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1000, 3, this));
        addVariable(FWSimVariable.createSimVariable("ivaporPressure", "vapor Pressure", DATA_TYPE.DOUBLE, CONTENT_TYPE.input,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1000, 6.1, this));
        addVariable(FWSimVariable.createSimVariable("iextraSolarRadiation", "extra Solar Radiation", DATA_TYPE.DOUBLE, CONTENT_TYPE.input,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1000, 11.7, this));
        addVariable(FWSimVariable.createSimVariable("netRadiation", " net radiation ", DATA_TYPE.DOUBLE, CONTENT_TYPE.out,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 5000, null, this));
        addVariable(FWSimVariable.createSimVariable("netOutGoingLongWaveRadiation", "net OutGoing Long Wave Radiation ", DATA_TYPE.DOUBLE, CONTENT_TYPE.out,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 5000, null, this));

        return iFieldMap;
    }
}

File: src/pycropml/transpiler/antlr_py/tests/examples/openalea/SQ_Energy_Balance/Netradiation.py
# coding: utf8
from copy import copy
from array import array
from math import *
from typing import *
from datetime import datetime

import numpy

#%%CyML Model Begin%%
def model_netradiation(minTair:float,
         maxTair:float,
         albedoCoefficient:float,
         stefanBoltzman:float,
         elevation:float,
         solarRadiation:float,
         vaporPressure:float,
         extraSolarRadiation:float):
    """
     - Name: NetRadiation -Version: 1.0, -Time step: 1
     - Description:
                 * Title: NetRadiation Model
                 * Authors: Peter D. Jamieson, Glen S. Francis, Derick R. Wilson, Robert J. Martin
                 * Reference:  https://doi.org/10.1016/0168-1923(94)02214-5
                 * Institution: New Zealand Institute for Crop and Food Research Ltd.,
                 New Zealand Institute for Crop and Food Research Ltd.,
                 New Zealand Institute for Crop and Food Research Ltd.,
                 New Zealand Institute for Crop and Food Research Ltd.
             
                 * ExtendedDescription: It is calculated at the surface of the canopy and is givenby the difference between incoming and outgoing radiation of both short
                          and long wavelength radiation 
                 * ShortDescription: It refers as difference between incoming and outgoing radiation of both short
                 and long wavelength radiation 
     - inputs:
                 * name: minTair
                               ** description : minimum air temperature
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** min : -30
                               ** max : 45
                               ** default : 0.7
                               ** unit : °C
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : variable
                 * name: maxTair
                               ** description : maximum air Temperature
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** min : -30
                               ** max : 45
                               ** default : 7.2
                               ** unit : °C
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : variable
                 * name: albedoCoefficient
                               ** description : albedo Coefficient
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** default : 0.23
                               ** min : 0
                               ** max : 1
                               ** unit : 
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : parameter
                 * name: stefanBoltzman
                               ** description : stefan Boltzman constant
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** default : 4.903E-09
                               ** min : 0
                               ** max : 1
                               ** unit : 
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : parameter
                 * name: elevation
                               ** description : elevation
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** default : 0
                               ** min : -500
                               ** max : 10000
                               ** unit : m
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : parameter
                 * name: solarRadiation
                               ** description : solar Radiation
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** default : 3
                               ** min : 0
                               ** max : 1000
                               ** unit : MJ m-2 d-1
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : variable
                 * name: vaporPressure
                               ** description : vapor Pressure
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** default : 6.1
                               ** min : 0
                               ** max : 1000
                               ** unit : hPa
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : variable
                 * name: extraSolarRadiation
                               ** description : extra Solar Radiation
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** default : 11.7
                               ** min : 0
                               ** max : 1000
                               ** unit : MJ m2 d-1
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                               ** inputtype : variable
     - outputs:
                 * name: netRadiation
                               ** description :  net radiation 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** min : 0
                               ** max : 5000
                               ** unit : MJ m-2 d-1
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
                 * name: netOutGoingLongWaveRadiation
                               ** description : net OutGoing Long Wave Radiation 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** min : 0
                               ** max : 5000
                               ** unit : g m-2 d-1
                               ** uri : http://www1.clermont.inra.fr/siriusquality/?page_id=547
    """

    netRadiation:float
    netOutGoingLongWaveRadiation:float
    Nsr:float
    clearSkySolarRadiation:float
    averageT:float
    surfaceEmissivity:float
    cloudCoverFactor:float
    Nolr:float
    Nsr = (1.0 - albedoCoefficient) * solarRadiation
    clearSkySolarRadiation = (0.75 + (2 * pow(10.0, -5) * elevation)) * extraSolarRadiation
    averageT = (pow(maxTair + 273.16, 4) + pow(minTair + 273.16, 4)) / 2.0
    surfaceEmissivity = 0.34 - (0.14 * sqrt(vaporPressure / 10.0))
    cloudCoverFactor = 1.35 * (solarRadiation / clearSkySolarRadiation) - 0.35
    Nolr = stefanBoltzman * averageT * surfaceEmissivity * cloudCoverFactor
    netRadiation = Nsr - Nolr
    netOutGoingLongWaveRadiation = Nolr
    return (netRadiation, netOutGoingLongWaveRadiation)
#%%CyML Model End%%